main.go

```diff
@@ -11,30 +14,95 @@ const (
 
 // generateRandomElements generates random elements.
 func generateRandomElements(size int) []int {
-	// ваш код здесь
+	if size <= 0 {
+		return []int{}
+	}
+	data := make([]int, size)
+	for i := range data {
+		data[i] = rand.Intn(1_000_000) + 1
+	}
+	return data
 }
 
 // maximum returns the maximum number of elements.
 func maximum(data []int) int {
-	// ваш код здесь
+	if len(data) == 0 {
+		return 0
+	}
+	max := data[0]
+	for _, val := range data[1:] {
+		if val > max {
+			max = val
+		}
+	}
+	return max
 }
 
 // maxChunks returns the maximum number of elements in a chunks.
 func maxChunks(data []int) int {
-	// ваш код здесь
+	if len(data) == 0 {
+		return 0
+	}
+
+	chunkSize := len(data) / CHUNKS
+	if chunkSize == 0 {
+		return maximum(data)
+	}
+
+	maximums := make([]int, CHUNKS)
+	var wg sync.WaitGroup
+	wg.Add(CHUNKS)
+
+	for i := 0; i < CHUNKS; i++ {
+		start := i * chunkSize
+		end := start + chunkSize
+		if i == CHUNKS-1 {
+			end = len(data)
+		}
+
+		if start >= len(data) {
+			maximums[i] = 0
+			wg.Done()
+			continue
+		}
+		if end > len(data) {
+			end = len(data)
+		}
+
+		go func(idx, s, e int) {
+			defer wg.Done()
+			maxVal := data[s]
+			for _, val := range data[s+1 : e] {
+				if val > maxVal {
+					maxVal = val
+				}
+			}
+			maximums[idx] = maxVal
```

Comment:
У тебя уже есть функция вычисления максимума
```suggestion
			maximums[idx] = maximum(data[s:e])
```